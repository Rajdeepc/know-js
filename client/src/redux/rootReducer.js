import { combineReducers } from "redux";
import AuthReducer from "../views/Auth/auth.reducer";
import NavbarReducer from '../components/Navbar/navbar.reducer'
import VideoPageReducer from '../views/VideoPage/video.reducer'
import HomePageReducer from '../views/Home/home.reducer'

const rootReducer =  combineReducers({
    AuthReducer,
    NavbarReducer,
    VideoPageReducer,
    HomePageReducer
})

export default rootReducer