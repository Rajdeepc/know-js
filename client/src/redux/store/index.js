import { createStore, applyMiddleware } from "redux"; // for creating store and applying middlerware like thunk
import thunk from "redux-thunk"; // for async actions
import rootReducer from "../rootReducer";
import { loadState, saveState } from "./sessionStorage";
import throttle from 'lodash/throttle';



const persistedState = loadState();

const store = createStore(
  rootReducer, 
  persistedState, 
  applyMiddleware(thunk));

store.subscribe(throttle(() => {
  saveState({
    LoginReducer: store.getState().LoginReducer
  });
},1000 ));


export default store;
