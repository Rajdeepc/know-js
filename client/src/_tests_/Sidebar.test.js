import React from "react";
import { render, cleanup } from "@testing-library/react";
import configureStore from "redux-mock-store";
import renderer from "react-test-renderer";
import SideBarComponent from "../components/Sidebar/Sidebar";
import "@testing-library/jest-dom/extend-expect";
import { Provider } from "react-redux";
import { createStore } from "redux";
import rootReducer from "../redux/rootReducer";
import { BrowserRouter as Router } from "react-router-dom";

afterEach(cleanup);

const store = createStore(rootReducer);

describe("Connected Redux <SideBarComponent component>", () => {
  const component = renderer.create(
    <Router>
      <Provider store={store}>
        <SideBarComponent />
      </Provider>
    </Router>
  );
  test("renders SideBarComponent Component", () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});
