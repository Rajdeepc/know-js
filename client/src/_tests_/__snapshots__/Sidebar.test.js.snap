// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Connected Redux <SideBarComponent component> renders SideBarComponent Component 1`] = `
<div
  className="sidebar-container"
>
  <div
    className="sidebar"
  >
    <div
      className="categories list-group"
      onKeyDown={[Function]}
    >
      <a
        className="list-group-item active"
        href="/datastructure"
        onClick={[Function]}
      >
        <span
          className="img-category"
        >
          <img
            alt="Data Structures"
            src="datastr.png"
          />
           
        </span>
        <span>
          Data Structures
        </span>
      </a>
      <a
        className="list-group-item"
        href="/javascript"
        onClick={[Function]}
      >
        <span
          className="img-category"
        >
          <img
            alt="JavaScript"
            src="javascript.png"
          />
           
        </span>
        <span>
          JavaScript
        </span>
      </a>
      <a
        className="list-group-item"
        href="/react"
        onClick={[Function]}
      >
        <span
          className="img-category"
        >
          <img
            alt="React"
            src="react.png"
          />
           
        </span>
        <span>
          React
        </span>
      </a>
      <a
        className="list-group-item"
        href="/redux"
        onClick={[Function]}
      >
        <span
          className="img-category"
        >
          <img
            alt="Redux"
            src="redux.png"
          />
           
        </span>
        <span>
          Redux
        </span>
      </a>
      <a
        className="list-group-item"
        href="/vue"
        onClick={[Function]}
      >
        <span
          className="img-category"
        >
          <img
            alt="Vue"
            src="vuejs.png"
          />
           
        </span>
        <span>
          Vue
        </span>
      </a>
      <a
        className="list-group-item"
        href="/svelte"
        onClick={[Function]}
      >
        <span
          className="img-category"
        >
          <img
            alt="Svelte"
            src={
              Object {
                "ReactComponent": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "render": [Function],
                },
                "__esModule": true,
                "default": "svelte.svg",
              }
            }
          />
           
        </span>
        <span>
          Svelte
        </span>
      </a>
      <a
        className="list-group-item"
        href="/angular"
        onClick={[Function]}
      >
        <span
          className="img-category"
        >
          <img
            alt="Angular"
            src="angular.png"
          />
           
        </span>
        <span>
          Angular
        </span>
      </a>
      <a
        className="list-group-item"
        href="/reactnative"
        onClick={[Function]}
      >
        <span
          className="img-category"
        >
          <img
            alt="React Native"
            src={
              Object {
                "ReactComponent": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "render": [Function],
                },
                "__esModule": true,
                "default": "reactnative.svg",
              }
            }
          />
           
        </span>
        <span>
          React Native
        </span>
      </a>
      <a
        className="list-group-item"
        href="/flutter"
        onClick={[Function]}
      >
        <span
          className="img-category"
        >
          <img
            alt="Flutter"
            src="flutter.png"
          />
           
        </span>
        <span>
          Flutter
        </span>
      </a>
      <a
        className="list-group-item"
        href="/mongodb"
        onClick={[Function]}
      >
        <span
          className="img-category"
        >
          <img
            alt="MongoDB"
            src={
              Object {
                "ReactComponent": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "render": [Function],
                },
                "__esModule": true,
                "default": "mongodb.svg",
              }
            }
          />
           
        </span>
        <span>
          MongoDB
        </span>
      </a>
      <a
        className="list-group-item"
        href="/nodejs"
        onClick={[Function]}
      >
        <span
          className="img-category"
        >
          <img
            alt="Node"
            src={
              Object {
                "ReactComponent": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "render": [Function],
                },
                "__esModule": true,
                "default": "nodejs.svg",
              }
            }
          />
           
        </span>
        <span>
          Node
        </span>
      </a>
    </div>
    <div
      className="my-videos list-group"
      onKeyDown={[Function]}
    />
  </div>
</div>
`;
